### PR TITLE
Feature/bubble size scaling

### PR DESCRIPTION
**Pull request by jseremba**

### CODE REVIEW

File: src/BubbleChart/BubbleChart.xml [hunk 1]
```diff
@@ -76,6 +76,16 @@
                         <attributeType name="Integer"/>
                     </attributeTypes>
                 </property>
+                <property key="autoBubbleSize" type="boolean" defaultValue="true">
+                    <caption>Auto bubble size</caption>
+                    <category>Data points</category>
+                    <description/>
+                </property>
+                <property key="markerSizeReference" type="integer" defaultValue="0">
```
Comment (Andries-Smit) [inline src/BubbleChart/BubbleChart.xml]: default would be 10

File: src/BubbleChart/BubbleChart.xml [hunk 2]
```diff
@@ -76,6 +76,16 @@
                         <attributeType name="Integer"/>
                     </attributeTypes>
```
Comment (Andries-Smit) [inline src/BubbleChart/BubbleChart.xml:77]: Rename `Bubble size data attribute` to `>Bubble size attribute`

File: src/BubbleChart/BubbleChart.xml [hunk 3]
```diff
@@ -76,6 +76,16 @@
...
                 </property>
+                <property key="autoBubbleSize" type="boolean" defaultValue="true">
+                    <caption>Auto bubble size</caption>
+                    <category>Data points</category>
+                    <description/>
+                </property>
+                <property key="markerSizeReference" type="integer" defaultValue="0">
+                    <caption>Marker size reference</caption>
```
Comment (Andries-Smit) [inline src/BubbleChart/BubbleChart.xml]: Rename to `Bubble size scale`

File: src/BubbleChart/BubbleChart.xml [hunk 4]
```diff
@@ -76,6 +76,16 @@
...
+                    <category>Data points</category>
+                    <description>The percentage to be used if the Auto bubble size is false</description>
```
Comment (Andries-Smit) [inline src/BubbleChart/BubbleChart.xml]: `markerSizeReference` should be used also for the auto.

 Description:
`When 'Auto bubble size' is true: The scale represent the percentage of the largest bubble. The largest bubble will cover the 10% of the canvas when the scale is set to 10.
When 'Auto bubble size' is false: The 'Bubble size' value is represented as the width of the bubble when the scale is set to 100 %`

File: src/LineChart/components/LineChart.ts [hunk 5]
```diff
@@ -274,4 +323,27 @@ export class LineChart extends Component<LineChartProps, LineChartState> {
...
         return traces;
     }
+
+    public static getMarkerSizeReference(props: LineChartProps, series: LineSeriesProps, markerSize: number[], dimensions?: { width: number; height: number }): number {
+        if (series.autoBubbleSize) {
+            const width = dimensions ? dimensions.width : 0;
+            const height = dimensions ? dimensions.height : 0;
+            let sizeRef = 0;
+            const averageSize = (width + height) / 2;
+            const percentageSize = averageSize / 10;
```
Comment (Andries-Smit) [inline src/LineChart/components/LineChart.ts]: Should use `markerSizeReference `
``` ts
const percentageSize = averageSize / series.markerSizeReference;
```

File: src/LineChart/components/LineChart.ts [hunk 6]
```diff
@@ -274,4 +323,27 @@ export class LineChart extends Component<LineChartProps, LineChartState> {
 
         return traces;
     }
+
+    public static getMarkerSizeReference(props: LineChartProps, series: LineSeriesProps, markerSize: number[], dimensions?: { width: number; height: number }): number {
+        if (series.autoBubbleSize) {
+            const width = dimensions ? dimensions.width : 0;
+            const height = dimensions ? dimensions.height : 0;
+            let sizeRef = 0;
```
Comment (Andries-Smit) [inline src/LineChart/components/LineChart.ts]: Default value for sizeRef should be `1`

File: src/LineChart/components/LineChart.ts [hunk 7]
```diff
@@ -274,4 +323,27 @@ export class LineChart extends Component<LineChartProps, LineChartState> {
...
+            const averageSize = (width + height) / 2;
+            const percentageSize = averageSize / 10;
+
+            if (markerSize.length > 0) {
+                sizeRef = Math.max(...markerSize) / percentageSize;
+            }
+
+            return Math.round(sizeRef * 1000) / 1000;
+        } else if (series.markerSizeReference > 0) {
```
Comment (Andries-Smit) [inline src/LineChart/components/LineChart.ts:343]: Integer modeler properties van not be < 0, so remove check
Or is this code called for non bubble charts too?

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: src/BubbleChart/BubbleChart.xml [hunk 8]
```diff
@@ -76,6 +76,16 @@
...
+                </property>
                 <property key="xValueSortAttribute" type="attribute" entityProperty="dataEntity" required="false" isPath="optional" pathType="reference">
                     <caption>X-axis sort attribute</caption>
                     <category>Data points</category>
```
Comment (Andries-Smit) [inline src/BubbleChart/BubbleChart.xml]: When updated the xml please check with Danny on confirmation of the bubble XML naming and description

File: src/BubbleChart/BubbleChart.xml (hunk 4 above)
Comment (Andries-Smit) [reply src/BubbleChart/BubbleChart.xml]: `When 'Auto bubble size' is true: The largest bubble will cover the 10% of the of the chart when the scale is set to 10. When 'Auto bubble size' is false: The 'Bubble size' is in pixels width multiplied by percentage of the scale`

Comment (edwinmp) [reply src/BubbleChart/BubbleChart.xml]: When 'Auto bubble size' is true, the largest bubble will cover the specified percentage of the chart. When 'Auto bubble size' is false, the 'Bubble size' is the specified percentage of the width in pixels.

File: src/BubbleChart/BubbleChart.xml (hunk 1 above)
Comment (edwinmp) [reply src/BubbleChart/BubbleChart.xml]: Done.

File: src/BubbleChart/BubbleChart.xml (hunk 2 above)
Comment (edwinmp) [reply src/BubbleChart/BubbleChart.xml:77]: Done.

File: src/BubbleChart/BubbleChart.xml (hunk 3 above)
Comment (edwinmp) [reply src/BubbleChart/BubbleChart.xml]: Done.

File: src/BubbleChart/BubbleChart.xml (hunk 4 above)
Comment (edwinmp) [reply src/BubbleChart/BubbleChart.xml]: `The percentage bubble size. When 'Auto bubble size' is true, the largest bubble will cover the specified percentage of the chart. When 'Auto bubble size' is false, the bubble size is the pixel value of the 'Bubble size attribute' multiplied by the percentage of the scale.`

File: src/LineChart/components/LineChart.ts (hunk 7 above)
Comment (edwinmp) [reply src/LineChart/components/LineChart.ts:343]: But it can be 0. The check is to avoid an Infinity result that arises from dividing by 0.

**Pull request closed without merging**